### ISSUE TITLE
Create a bot to track blocked PRs

### ISSUE THREAD

**Issue by pietroalbini** [rust-ops/requests#2]
Currently blocked PRs are periodically checked to make sure they're still blocked, or to remove the `S-blocked` label. I propose to create a bot to automatically track that (at least simple cases, like a PR blocked by another PR).

## Proposed behavior

When someone with the right permissions types this in a comment:

```
@rust-highfive blocked #12345
```

...the bot will apply the `S-blocked` label, remove all the other `S-` ones and store the PR in a database. Then, when the bot receives a webhook saying the linked issue/PR was closed, it will reopen the PR and apply the `S-waiting-on-author` label, removing it from the database.
